src/test/java/com/algolia/search/integration/client/SecuredAPIKeyTest.java

```diff
@@ -69,5 +52,9 @@ void testSecuredAPIKey() throws Exception {
 
     assertThatThrownBy(() -> index2WithRestriction.search(new Query()))
         .hasMessageContaining("Index not allowed with this API key");
+
+    clientWithRestriction.close();
+
+    index2.deleteAsync().get().waitTask();
```

Comment:
Should we consider putting back the `deleteAsync()` in the `@AfterAll` method? Or maybe within a `try/catch/finally` block, because If something wrong happens the index is not deleted.

Moreover, the `waitTask()` is not necessary as this deletion is a fire&forget. (Usually we perform a waitTask when we want to be sure that the task has been processed by the engine or when the next steps are dependents on this task).

Regarding the `get()`, Will the HTTP call be sent correctly if we remove it? I think so, WDYT? 🤔

Reply:
> Should we consider putting back the deleteAsync() in the @AfterAll method? Or maybe within a try/catch/finally block, because If something wrong happens the index is not deleted.
I would suggest going for a try/catch/finally block, to keep the index variable local (I realise that the index itself is global, but as much as possible I would rather keep variables locals). Will update the comment.

> `waitTask()` is not necessary as the deletion is fire&forget.
Agreed, the deletion is fire&forget, so `waitTask`, in such a test, is not necessary.

On the other hand, tests should be as illustrative of best practices as possible: I wouldn't blame a customer of ours copy/pasting a snippet of code found in tests. And if I understand correctly, we invite customers to check if not wait the result of tasks.

Let's start by staying closer to the original code, by not having `get` and `waitTask`, and see for the best-practice thing later on.

> Will the HTTP call be sent if we remove the `get()`
It depends on when the parallel execution starts - on when the future is actually triggering a new thread. Have to check that. Let's say that if it doesn't, it's a bug (existing in the current "AfterAll"-based code base), requiring a bug fix on its own (ie in a separated PR).

Reply:
> I would suggest going for a try/catch/finally block, to keep the index variable local (I realise that the index itself is global, but as much as possible I would rather keep variables locals). Will update the comment.

👍 

> It depends on when the parallel execution starts - on when the future is actually triggering a new thread. I have the feeling that it happens _before_ the HTTP request is effectively done ^_^; I have to check that. Let's say that if it doesn't, it's a bug (existing in the current "AfterAll"-based code base), requiring a bug fix on its own (ie in a separated PR).

Let's put belt and braces and keep the `get()` 😄